src/components/App.tsx

```diff
@@ -2260,10 +2298,15 @@ class App extends React.Component<any, AppState> {
         });
         document.documentElement.style.cursor = CURSOR_TYPE.POINTER;
       } else {
+        const [gridX, gridY] = getGridPoint(
+          x,
+          y,
+          this.state.elementType === "draw" ? null : this.state.gridSize,
```

Comment:
I don't think this is very good in readability. Feel free to refactor...

Reply:
It's fine. Will be removed when we add support for all types later, anyway.